### PR TITLE
small backend changes for deleting email contacts

### PR DESCRIPTION
**Pull request by martinraison**
@stkem

### CODE REVIEW

**Comment by stkem** [comment]
@lawzlo FYI

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/ext/ExtNonUserSearchController.scala [hunk 1]
```diff
@@ -35,11 +37,13 @@ class ExtNonUserSearchController @Inject() (
...
 
-  def hideEmailFromUser(email: String) = JsonAction.authenticatedAsync { request =>
-    new SafeFuture[Boolean]({
-      typeaheadCommander.hideEmailFromUser(request.userId, EmailAddress(email))
-    }) map { result =>
-      Ok(Json.toJson(result))
-    }
+  def hideEmailFromUser() = JsonAction.authenticatedParseJsonAsync { request =>
+    (request.body \ "email").asOpt[String] map { email =>
+      new SafeFuture[Boolean]({
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/ext/ExtNonUserSearchController.scala:42]: This is not safe, we should be using as.Opt[EmailAddress] for proper validation the the address.

Comment (martinraison) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/ext/ExtNonUserSearchController.scala:42]: Oh, awesome - changing that now

**Comment by lawzlo** [comment]
Nice error handling.

**Pull request merged**